server/application/application.go

```diff
@@ -1333,9 +1333,16 @@ func (s *Server) getAppResources(ctx context.Context, a *appv1.Application) (*ap
 }
 
 func (s *Server) getAppLiveResource(ctx context.Context, action string, q *application.ApplicationResourceRequest) (*appv1.ResourceNode, *rest.Config, *appv1.Application, error) {
+	enableV3, err := s.settingsMgr.GetServerRBACEnableV3()
```

Comment:
`GetServerRBACEnableV3` is very generic and unclear what this does. It also introduces arbitrary code that this will be implemented in v3, which is unknown. 

I think we should call this `disableApplicationFineGrainedRBACInheritance=true`. It is a mouthful, but it is clearer what it does and we can Change the default in v3 (https://github.com/argoproj/argo-cd/issues/19988) and remove it in v4.